LoopKit/Security/Security.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  Security.swift
+//  LoopKit
+//
+//  Created by Nathaniel Hamming on 2023-09-05.
+//  Copyright © 2023 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+public protocol SecurityDelegate: AnyObject {
+    /// Informs the delegate that the state of the specified security was updated and the delegate should persist the security. May
+    /// be invoked prior to the security completing setup.
+    ///
+    /// - Parameters:
+    ///     - security: The security that updated state.
+    func securityDidUpdateState(_ security: Security)
+}
+
+public protocol SecurityPlugin {
+    var securityType: Security.Type? { get }
+}
+
+public protocol SecurityProvider {
+    /// The security with the specified identifier.
+    ///
+    /// - Parameters:
+    ///     - identifier: The identifier of the security
+    /// - Returns: Either a security with matching identifier or nil.
+    func security(withIdentifier identifier: String) -> Security?
+}
+
+public protocol Security: Pluggable {
```

Comment:
This describes a Pluggable that has persistence, and has nothing "security" related. Could we change the name to reflect that? I could see other plugins vending a pluggable that needs persistence. Maybe `PersistedPluggable`? or `StatefulPluggable`?

Reply:
@ps2 updated and made adjustments to `Service` to make sure of this.